src/pretalx/agenda/views/talk.py

```diff
@@ -66,14 +66,14 @@ class TalkView(PermissionRequired, DetailView):
 
     def get_object(self, queryset=None):
         with suppress(AttributeError, TalkSlot.DoesNotExist):
-            return self.request.event.current_schedule.talks.get(
+            return self.request.event.current_schedule.talks.filter(
                 submission__code__iexact=self.kwargs['slug'], is_visible=True
-            )
+            ).first()
         if getattr(self.request, 'is_orga', False):
             with suppress(AttributeError, TalkSlot.DoesNotExist):
-                return self.request.event.wip_schedule.talks.get(
+                return self.request.event.wip_schedule.talks.filter(
                     submission__code__iexact=self.kwargs['slug'], is_visible=True
-                )
+                ).first()
```

Comment:
> Wouldn't it be better to make the submission itself the object of the template and view, now that we potentially have multiple slots?
sounds good -  i have tried this but it head some side-effects /errors on other code i currently not familiar with - (permission things)
so i prepared this in the `get_context_data` function but used the `filter(..).first()` way around..